api/libproject/devstart.rb

```diff
@@ -173,34 +178,51 @@ def dev_up()
-  common.run_inline %W{docker-compose run db-scripts ./run-migrations.sh main}
-  init_new_cdr_db %W{--cdr-db-name cdr}
 
-  common.status "Updating CDR versions..."
-  common.run_inline %W{docker-compose run api-scripts ./gradlew updateCdrVersions -PappArgs=['/w/api/config/cdr_versions_local.json',false]}
+  overall_bm = Benchmark.measure {
+    common.status "Database startup..."
+    bm = Benchmark.measure { common.run_inline %W{docker-compose up -d db} }
+    common.status "Database startup complete (#{format_benchmark(bm)})"
 
-  common.status "Updating workbench configuration..."
-  common.run_inline %W{
-    docker-compose run api-scripts
-    ./gradlew loadConfig -Pconfig_key=main -Pconfig_file=config/config_local.json
-  }
-  common.status "Updating CDR schema configuration..."
-  common.run_inline %W{
-    docker-compose run api-scripts
-    ./gradlew loadConfig -Pconfig_key=cdrBigQuerySchema -Pconfig_file=config/cdm/cdm_5_2.json
-  }
-  common.status "Updating featured workspaces..."
-  common.run_inline %W{
-    docker-compose run api-scripts
-    ./gradlew loadConfig -Pconfig_key=featuredWorkspaces -Pconfig_file=config/featured_workspaces_local.json
-  }
+    common.status "Database init & migrations..."
+    bm = Benchmark.measure {
+      common.run_inline %W{docker-compose run db-scripts ./run-migrations.sh main}
+      init_new_cdr_db %W{--cdr-db-name cdr}
+    }
+    common.status "Database init & migrations complete (#{format_benchmark(bm)})"
+
+    common.status "Pushing configs & data..."
+    bm = Benchmark.measure {
+      common.status "  --> Updating CDR versions"
+      common.run_inline %W{docker-compose run api-scripts ./gradlew updateCdrVersions -PappArgs=['/w/api/config/cdr_versions_local.json',false]}
+
+      common.status "  --> Loading Workbench config"
+      common.run_inline %W{
+        docker-compose run api-scripts
+        ./gradlew loadConfig -Pconfig_key=main -Pconfig_file=config/config_local.json
+      }
+
+      common.status "  --> Loading CDR BigQuery Schema config"
+      common.run_inline %W{
+        docker-compose run api-scripts
+        ./gradlew loadConfig -Pconfig_key=cdrBigQuerySchema -Pconfig_file=config/cdm/cdm_5_2.json
+      }
+
+      common.status "  --> Loading Featured Workspaces config"
+      common.run_inline %W{
+        docker-compose run api-scripts
+        ./gradlew loadConfig -Pconfig_key=featuredWorkspaces -Pconfig_file=config/featured_workspaces_local.json
+      }
+
+      common.status "  --> Loading Data Dictionary data"
+      common.run_inline %W{docker-compose run api-scripts ./gradlew loadDataDictionary -PappArgs=false}
+    }
+    common.status "Pushing configs complete (#{format_benchmark(bm)})"
 
-  common.status "Loading Data Dictionary..."
-  common.run_inline %W{docker-compose run api-scripts ./gradlew loadDataDictionary -PappArgs=false}
+  }
+  common.status "All dev-env setup steps complete (#{format_benchmark(overall_bm)})"
```

Comment:
opt: `(total: #{format_benchmark(overall_bm)})` in parens to differentiate from the other steps, which are all individual runtimes? Scanning the output I thought the last two steps each took ~500s.

Reply:
Good suggestion – I updated to more clearly reflect that it's counting the total time.